packages/babel-plugin-transform-remove-console/src/index.js

```diff
@@ -1,14 +1,67 @@
 "use strict";
 
-module.exports = function() {
+module.exports = function({ types: t }) {
   return {
     name: "transform-remove-console",
     visitor: {
       CallExpression(path) {
-        if (path.get("callee").matchesPattern("console", true)) {
-          path.remove();
+        const callee = path.get("callee");
+
+        if (!callee.isMemberExpression()) return;
+
+        if (isConsole(callee)) {
+          // console.log()
+          if (path.parentPath.isExpressionStatement()) {
+            path.remove();
+          } else {
+            path.replaceWith(createVoid0());
+          }
+        } else if (isConsoleBind(callee)) {
+          // console.log.bind()
+          path.replaceWith(createNoop());
         }
       },
+      MemberExpression: {
+        exit(path) {
+          if (isConsole(path) && !path.parentPath.isMemberExpression()) {
+            path.replaceWith(createNoop());
+          }
```

Comment:
It looks not work with `console.err = xx` ([React Native have similar usage](https://github.com/facebook/react-native/blob/master/Libraries/Core/ExceptionsManager.js#L103-L104)), maybe we can just `parentPath.remove()` if path key is `left` and parentPath is assignment expression?